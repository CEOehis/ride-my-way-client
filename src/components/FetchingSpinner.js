import React from 'react';

const FetchingSpinner = () => (
  <i style={{ textAlign: 'center', display: 'block' }} className="fa fa-spinner fa-spin fa-pulse fa-5x" aria-hidden="true" />
);

export default FetchingSpinner;
