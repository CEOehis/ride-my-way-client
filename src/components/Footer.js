import React from 'react';

const Footer = () => (
  <footer className="app-grid orange-bg">
    <div className="layout-grid">
      <p>Â© 2018. Hand crafted by Celestine Ekoh-Ordan</p>
    </div>
  </footer>
);

export default Footer;
