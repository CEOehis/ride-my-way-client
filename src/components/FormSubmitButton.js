import React from 'react';
import PropTypes from 'prop-types';

const FormSubmitButton = ({ submitting }) => (submitting
  ? <input className="btn btn-lg btn-submit btn-orange submitting" type="submit" value="submitting" disabled />
  : <input className="btn btn-lg btn-submit btn-orange" type="submit" value="submit" />);


FormSubmitButton.propTypes = {
  submitting: PropTypes.bool.isRequired,
};

export default FormSubmitButton;
