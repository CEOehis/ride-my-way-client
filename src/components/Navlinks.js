import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const Navlinks = ({ user, isAuthenticated, signOut }) => {
  if (isAuthenticated) {
    return (
      <Fragment>
        <li><Link className="brand" to="/">RideMyWay</Link></li>
        <li><Link to="/create">Offer a Ride</Link></li>
        <li><Link to="/home">View Rides</Link></li>
        <li className="dropdown">
          <a>
            {
              user.fullName
                ? `Hi, ${user.fullName.split(' ')[0]}`
                : <i className="fa fa-spinner fa-spin fa-pulse" />
            }
            <i className="fa fa-chevron-down" />
          </a>
          <ul className="dropdown-content">
            <li><Link to="/profile">Profile</Link></li>
            <li><Link to="/user/rides">My Ride Offers</Link></li>
            <li><a onClick={signOut}>Sign out</a></li>
          </ul>
        </li>
      </Fragment>
    );
  }

  return (
    <Fragment>
      <li>
        <Link to="/" className="brand">RideMyWay</Link>
      </li>
      <li>
        <Link to="/register">Register</Link>
      </li>
      <li>
        <Link to="/login">Login</Link>
      </li>
    </Fragment>
  );
};

Navlinks.propTypes = {
  user: PropTypes.shape({
    email: PropTypes.string,
    fullName: PropTypes.string,
    phone: PropTypes.string,
    userId: PropTypes.number,
  }),
  signOut: PropTypes.func,
  isAuthenticated: PropTypes.bool.isRequired,
};

Navlinks.defaultProps = {
  signOut: () => null,
};

Navlinks.defaultProps = {
  user: undefined,
};

export default Navlinks;
