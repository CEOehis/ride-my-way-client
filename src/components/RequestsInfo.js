import React, { Fragment } from 'react';

const RequestsInfo = ({ error, requests, handleResponse }) => {
  if (error) {
    return (
      <div className="request-summary">
        Unable to fetch requests information at this time
      </div>
    );
  }

  return (
    requests.map(request => (
      <div key={request.requestId} className="request-summary">
        <div className="profile">
          <img src="https://placehold.it/64/64" alt="" />
          <div className="driver">
            <h4>{request.requesterName.split(' ')[0]}</h4>
          </div>
        </div>
        <div className="actions">
          {
            request.offerStatus === 'pending'
              ? (
                <Fragment>
                  <button
                    type="button"
                    data-request-id={request.requestId}
                    data-ride-id={request.rideId}
                    data-action="accept"
                    className="btn btn-orange"
                    onClick={handleResponse}
                  >
                    Accept
                  </button>
                  <button
                    type="button"
                    data-request-id={request.requestId}
                    data-ride-id={request.rideId}
                    data-action="reject"
                    className="btn btn-orange-inverse"
                    onClick={handleResponse}
                  >
                    Reject
                  </button>
                </Fragment>
              )
              : (
                <span>
                  Request
                  {' '}
                  {request.offerStatus}
                </span>
              )
          }
        </div>
      </div>
    ))
  );
};

export default RequestsInfo;
