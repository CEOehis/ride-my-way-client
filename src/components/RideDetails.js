/* eslint-disable no-nested-ternary */
import React, { Fragment } from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import parseTime from '../utils/parseTime';

const RideDetails = ({
  ride,
  user,
  requested,
  handleRequest,
  requesting,
}) => {
  const {
    origin,
    destination,
    departureDate,
    departureTime,
    seats,
    rideId,
    rideCreator,
    email,
    phone,
  } = ride;
  if (!Object.keys(ride).length) return null;
  return (
    <Fragment>
      <h1>
        {origin}
        {' '}
        to
        {' '}
        {destination}
      </h1>
      <div className="ride-details">
        <div className="trip">
          <h1>Trip information</h1>
          <table>
            <tbody>
              <tr>
                <td>From: </td>
                <td>
                  {origin}
                </td>
              </tr>
              <tr>
                <td>To: </td>
                <td>
                  {destination}
                </td>
              </tr>
              <tr>
                <td>Departing: </td>
                <td>
                  {moment(departureDate).format('MMM DD, YYYY')}
                  ,
                  {parseTime(departureTime)}
                </td>
              </tr>
              <tr>
                <td>Available seats: </td>
                <td>
                  {seats}
                </td>
              </tr>
              <tr>
                <td>status: </td>
                <td>Available</td>
              </tr>
            </tbody>
          </table>
          <div className="call-to-action">
            {user.email !== email
              ? (
                <button
                  className={`btn btn-orange btn-lg ${requesting || requested ? 'submitting' : ''}`}
                  data-ride-id
                  href={rideId}
                  type="button"
                  disabled={requested}
                  onClick={handleRequest}
                >
                  {
                    requesting
                      ? 'requesting'
                      : requested ? 'request pending' : 'Join this Ride'
                  }
                </button>
              ) : ''
  }
            <Link className="btn btn-orange-inverse btn-lg" to="/home">More Rides</Link>
          </div>
        </div>
        <div className="driver-profile">
          <h1>Driver Profile</h1>
          <img src={`http://i.pravatar.cc/150?u=${email}`} alt="host-img" />
          <h2 className="driver-name">
            {rideCreator}
          </h2>
          <hr style={{ border: '3px solid #f1f1f1' }} />
          <p>
            email:
            {' '}
            {email}
          </p>
          <p>
            phone:
            {' '}
            {phone || '+234'}
          </p>
          <hr style={{ border: '3px solid #f1f1f1' }} />
        </div>
      </div>
    </Fragment>
  );
};

RideDetails.propTypes = {
  ride: PropTypes.object.isRequired,
  user: PropTypes.object.isRequired,
  requested: PropTypes.bool.isRequired,
  requesting: PropTypes.bool.isRequired,
  handleRequest: PropTypes.func.isRequired,
};

const mapStateToProps = ({ auth, rideDetails }) => ({
  user: auth.user,
  requested: rideDetails.requested,
  requesting: rideDetails.requesting,
});

export default connect(mapStateToProps)(RideDetails);
