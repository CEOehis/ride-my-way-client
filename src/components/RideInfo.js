import React from 'react';
import PropTypes from 'prop-types';

const RideInfo = ({ error, ride: { origin, destination } }) => {
  if (error) {
    return (
      <React.Fragment>
        <h2>Ride information</h2>
        <h3>Unable to fetch ride information at this time</h3>
      </React.Fragment>
    );
  }

  return (
    <React.Fragment>
      <h2>Ride information</h2>
      <h3>
        {origin}
        {' '}
        <span>
          <i className="fa fa-arrow-circle-o-right" aria-hidden="true" />
        </span>
        {' '}
        {destination}
      </h3>
    </React.Fragment>
  );
};

RideInfo.propTypes = {
  error: PropTypes.string,
  ride: PropTypes.object.isRequired,
};

RideInfo.defaultProps = {
  error: '',
};

export default RideInfo;
