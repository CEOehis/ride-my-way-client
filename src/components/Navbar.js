import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Navlinks from './Navlinks';
import signOut from '../actions/signOut';

export class Navbar extends React.Component {
  constructor() {
    super();

    this.state = {
      mobileNavVisible: false,
    };

    this.toggleMenu = this.toggleMenu.bind(this);
    this.handleSignOut = this.handleSignOut.bind(this);
  }

  toggleMenu() {
    const { mobileNavVisible } = this.state;
    this.setState({
      mobileNavVisible: !mobileNavVisible,
    });
  }

  handleSignOut() {
    const { logOut } = this.props;
    logOut();
  }

  render() {
    const { mobileNavVisible } = this.state;
    const user = JSON.parse(localStorage.getItem('user'));
    return (
      <nav className="layout-grid navbar">
        <h1 className="brand"><Link to="/">RideMyWay</Link></h1>
        <button onClick={this.toggleMenu} className="menu btn" type="button">
          <i className="fa fa-navicon" aria-hidden="true" />
        </button>
        <ul className={mobileNavVisible ? 'nav-links expanded' : 'nav-links'}>
          <Navlinks user={user} signOut={this.handleSignOut} />
        </ul>
      </nav>
    );
  }
}

Navbar.propTypes = {
  logOut: PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => ({
  logOut() {
    dispatch(signOut());
  },
});

export default connect(null, mapDispatchToProps)(Navbar);
