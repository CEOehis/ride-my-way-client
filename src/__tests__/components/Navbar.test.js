import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Navbar } from '../../components/Navbar';

describe('Form submit button component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Navbar />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
