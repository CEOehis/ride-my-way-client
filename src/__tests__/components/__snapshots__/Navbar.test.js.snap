// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form submit button component should render correctly 1`] = `
<nav
  className="layout-grid navbar"
>
  <h1
    className="brand"
  >
    <Link
      replace={false}
      to="/"
    >
      RideMyWay
    </Link>
  </h1>
  <button
    className="menu btn"
    onClick={[Function]}
    type="button"
  >
    <i
      aria-hidden="true"
      className="fa fa-navicon"
    />
  </button>
  <ul
    className="nav-links"
  >
    <Navlinks
      signOut={[Function]}
      user={null}
    />
  </ul>
</nav>
`;
